From ac4074f4080398c9801ac4724453f0323bae04b5 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 4 Jul 2024 18:49:59 +0200
Subject: [PATCH 1168/1695] dt-bindings: gpio: stm32mp: flags for non-secure
 GPIOs

Define STM32 GPIO DT bindings bit flags for GPIOs that are to be used
in non-secure state.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 73f95e41f81eca4e2a9e6c25d0f6b4ef366eccda dt-bindings: gpio: stm32mp: flags for non-secure GPIOs]
Change-Id: Ia0b6717ebdc9b6ec5748da418557617d8f3efecd
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391590
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/include/dt-bindings/gpio/stm32mp_gpio.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/include/dt-bindings/gpio/stm32mp_gpio.h b/core/include/dt-bindings/gpio/stm32mp_gpio.h
index ea0466d28..ba3cd34ab 100644
--- a/core/include/dt-bindings/gpio/stm32mp_gpio.h
+++ b/core/include/dt-bindings/gpio/stm32mp_gpio.h
@@ -11,4 +11,7 @@
 /* Macro to define the security for GPIO */
 #define TZPROT(id) (1 << (id))
 
+/* GPIO phandle argument bitmask for a non-secure GPIO */
+#define GPIO_STM32_NSEC		(1 << 31)
+
 #endif
-- 
2.47.2

